sdk/inc/azure/core/az_mqtt.h

```diff
@@ -0,0 +1,391 @@
+  AZ_MQTT_X509_CLIENT_CERTIFICATE_KEY_SECURITY_MODULE = 1,
+} az_mqtt_x509_client_certificate_key_type;
+
+/**
+ * @brief x509 certificate definition.
+ */
+typedef struct
+{
+  az_span cert;
+  az_span key;
+  az_mqtt_x509_client_certificate_key_type key_type;
+} az_mqtt_x509_client_certificate;
+
+/**
+ * @brief MQTT publish data.
+ *
+ */
+typedef struct
+{
+  az_span topic;
+  az_span payload;
+
+  // The MQTT stack should set this ID upon returning.
+  int32_t out_id;
+  int8_t qos;
+} az_mqtt_pub_data;
+
+/**
+ * @brief MQTT receive data.
+ *
+ */
+typedef struct
+{
+  az_span topic;
+  az_span payload;
+  int8_t qos;
+  int32_t id;
+} az_mqtt_recv_data;
+
+/**
+ * @brief MQTT publish acknowledgement data.
+ *
+ */
+typedef struct
+{
+  int32_t id;
+} az_mqtt_puback_data;
+
+/**
+ * @brief MQTT subscribe data.
+ *
+ */
+typedef struct
+{
+  az_span topic_filter;
+  int8_t qos;
+  int32_t out_id;
+} az_mqtt_sub_data;
+
+/**
+ * @brief MQTT subscribe acknowledgement data.
+ *
+ */
+typedef struct
+{
+  int32_t id;
+} az_mqtt_suback_data;
+
+/**
+ * @brief MQTT connect data.
+ *
+ */
+typedef struct
+{
+  az_span host;
+  int16_t port;
+  az_span username;
+  az_span password;
+  az_span client_id;
+
+  az_mqtt_x509_client_certificate certificate;
+} az_mqtt_connect_data;
+
+/**
+ * @brief MQTT connect acknowledgement data.
+ *
+ */
+typedef struct
+{
+  int32_t connack_reason;
+  bool tls_authentication_error;
+} az_mqtt_connack_data;
+
+/**
+ * @brief MQTT disconnect data.
+ *
+ */
+typedef struct
+{
+  bool tls_authentication_error;
+  bool disconnect_requested;
+} az_mqtt_disconnect_data;
+
+enum az_log_classification_mqtt
+{
+  AZ_LOG_MQTT_STACK = _az_LOG_MAKE_CLASSIFICATION(_az_FACILITY_CORE_MQTT, 3),
+};
+
+/**
+ * @brief Azure MQTT HFSM event types.
+ *
+ */
+enum az_event_type_mqtt
+{
+  /// MQTT Connect Request event.
+  AZ_MQTT_EVENT_CONNECT_REQ = _az_MAKE_EVENT(_az_FACILITY_CORE_MQTT, 10),
+
+  /// MQTT Connect Response event.
+  AZ_MQTT_EVENT_CONNECT_RSP = _az_MAKE_EVENT(_az_FACILITY_CORE_MQTT, 11),
+
+  /// MQTT Disconnect Request event.
+  AZ_MQTT_EVENT_DISCONNECT_REQ = _az_MAKE_EVENT(_az_FACILITY_CORE_MQTT, 12),
+
+  /// MQTT Disconnect Response event.
+  AZ_MQTT_EVENT_DISCONNECT_RSP = _az_MAKE_EVENT(_az_FACILITY_CORE_MQTT, 13),
+
+  /// MQTT Publish Receive Indication event.
+  AZ_MQTT_EVENT_PUB_RECV_IND = _az_MAKE_EVENT(_az_FACILITY_CORE_MQTT, 14),
+
+  /// MQTT Publish Request event.
+  AZ_MQTT_EVENT_PUB_REQ = _az_MAKE_EVENT(_az_FACILITY_CORE_MQTT, 15),
+
+  /// MQTT Publish Acknowledge Response event.
+  AZ_MQTT_EVENT_PUBACK_RSP = _az_MAKE_EVENT(_az_FACILITY_CORE_MQTT, 16),
+
+  /// MQTT Subscribe Request event.
+  AZ_MQTT_EVENT_SUB_REQ = _az_MAKE_EVENT(_az_FACILITY_CORE_MQTT, 17),
+
+  /// MQTT Subscribe Acknowledge Response event.
+  AZ_MQTT_EVENT_SUBACK_RSP = _az_MAKE_EVENT(_az_FACILITY_CORE_MQTT, 18),
+};
+
+enum
+{
+  AZ_IOT_DEFAULT_MQTT_CONNECT_KEEPALIVE_SECONDS = 240
```

Comment:
should this use the value from sdk/inc/azure/core/az_config.h (or vice versa)?

Reply:
Moved values to `az_mqtt_config.h` as per disucssions